### PR TITLE
[DEV-10056] Add State/County Fips and County Name to Subaward Downloads

### PR DESCRIPTION
**Pull request by dporth-frb**
**Description:**
Another feature added two new columns to PrimeAward and PrimeTransaction downloads. These columns represented some new data about FIPS. The purpose of this work and PR is to make the same or similar data about FIPS available in the Subaward downloads. 

**Technical details:**
Details about new columns added:
* legal_entity_state_fips: 2-character recipient state FIPS code of the parent prime award summary.
* place_of_perform_state_fips: 2-character place of performance state FIPS code of the parent prime award summary.
* Deriving legal_entity and place_of_perform county zips
   * For county derivations, we needed to match county names and state codes to `ref_city_county_state_code` to obtain the 3-digit county fips.
* How does subaward_search access county name?
   * Subawards needs county name to include in the join to `ref_city_county_state_code`. The logic to obtain county name in this PR matches that in transaction search and award search.

Other Notes:
* This PR requires data change tickets. The relevant tickets have already been created and documented in the deploy coordination for the sprint.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [x] Necessary PR reviewers:
    - [x] Backend
    - [x] Frontend <OPTIONAL> (N/A)
    - [x] Operations <OPTIONAL> (N/A)
    - [x] Domain Expert <OPTIONAL> (N/A)
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-10056](https://federal-spending-transparency.atlassian.net/browse/DEV-10056):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
3b, 3c, 3d The nature of this work does not require PR reviews from these groups
4. The nature of this work does not impact materialized views
5. The nature of this work does not require an impact assessment from the frontend

### CODE REVIEW

File: usaspending_api/search/delta_models/subaward_search.py [hunk 1]
```diff
@@ -494,8 +523,20 @@
...
-        END) AS sub_place_of_performance_congressional_current
-
+        END) AS sub_place_of_performance_congressional_current,
+        rl_state_fips.fips as legal_entity_state_fips,
+        pop_state_fips.fips as place_of_perform_state_fips,
+        case
+            when rl_state_fips.fips is null or rl_county_fips.county_numeric is null
+                then NULL
+            else concat(rl_state_fips.fips, rl_county_fips.county_numeric)
+        end as legal_entity_county_fips,
```
Comment (collinwr) [inline usaspending_api/search/delta_models/subaward_search.py:533]: These case statements are unnecessary. In SparkSQL, if any of the inputs to a CONCAT function are NULL, the output will be null.

Comment (dporth-frb) [reply usaspending_api/search/delta_models/subaward_search.py:533]: Ah good catch! Thank you.

File: usaspending_api/search/delta_models/subaward_search.py [hunk 2]
```diff
@@ -253,6 +258,30 @@
             faba.award_id IS NOT NULL
         GROUP BY
             faba.award_id
+    ),
+    state_fips (
+        select
```
Comment (collinwr) [inline usaspending_api/search/delta_models/subaward_search.py:263]: Can you please capitalize the SQL keywords throughout this file? (`SELECT`, `FROM`, `GROUP`, `BY`, `AS`, etc) to be more consistent with the file. I will add a some documentation around this in our SQL style guide.

Comment (dporth-frb) [reply usaspending_api/search/delta_models/subaward_search.py:263]: Ok that'd be great to have this expectation documented.

Comment (dporth-frb) [reply usaspending_api/search/delta_models/subaward_search.py:263]: Can you link the style guide?

**Review by collinwr — APPROVED** [review]

**Pull request merged**